#### Part 1: Exploration and Data Quality

In [7]:
# Read the raw data files and summarize them
from pathlib import Path
import pandas as pd

RAW = Path("../data/raw")

FILES = {
    "orders": "olist_orders_dataset.csv",
    "items": "olist_order_items_dataset.csv",
    "payments": "olist_order_payments_dataset.csv",
    "reviews": "olist_order_reviews_dataset.csv",
    "customers": "olist_customers_dataset.csv",
    "sellers": "olist_sellers_dataset.csv",
    "products": "olist_products_dataset.csv",
    "geo": "olist_geolocation_dataset.csv",
    "cat_en": "product_category_name_translation.csv",
}

d = {k: pd.read_csv(RAW / v) for k, v in FILES.items()}

summary = pd.DataFrame(
    [
        {
            "table": k,
            "filename": FILES[k],
            "rows": v.shape[0],
            "n_cols": v.shape[1],
            "columns": ", ".join(v.columns),
        }
        for k, v in d.items()
    ]
).set_index("table")

pd.set_option("display.max_colwidth", None)
summary

,filename,rows,n_cols,columns
table,,,,
orders,olist_orders_dataset.csv,99441,8,"order_id, customer_id, order_status, order_purchase_timestamp, order_approved_at, order_delivered_carrier_date, order_delivered_customer_date, order_estimated_delivery_date"
items,olist_order_items_dataset.csv,112650,7,"order_id, order_item_id, product_id, seller_id, shipping_limit_date, price, freight_value"
payments,olist_order_payments_dataset.csv,103886,5,"order_id, payment_sequential, payment_type, payment_installments, payment_value"
reviews,olist_order_reviews_dataset.csv,99224,7,"review_id, order_id, review_score, review_comment_title, review_comment_message, review_creation_date, review_answer_timestamp"
customers,olist_customers_dataset.csv,99441,5,"customer_id, customer_unique_id, customer_zip_code_prefix, customer_city, customer_state"
sellers,olist_sellers_dataset.csv,3095,4,"seller_id, seller_zip_code_prefix, seller_city, seller_state"
products,olist_products_dataset.csv,32951,9,"product_id, product_category_name, product_name_lenght, product_description_lenght, product_photos_qty, product_weight_g, product_length_cm, product_height_cm, product_width_cm"
geo,olist_geolocation_dataset.csv,1000163,5,"geolocation_zip_code_prefix, geolocation_lat, geolocation_lng, geolocation_city, geolocation_state"
cat_en,product_category_name_translation.csv,71,2,"product_category_name, product_category_name_english"


##### The raw dataset consists of 9 tables:
1. Orders (99,441 rows × 8 columns): Contains order-level information, including order ID, customer ID, order status, and order timestamps.
2. Items (112,650 rows × 7 columns): Contains details of items in each order, including order ID, product ID, seller ID, shipping limit date, price, and freight value.
3. Payments (103,886 rows × 5 columns): Contains payment details for each order, including payment type, payment installments, payment sequence, and payment value.
4. Reviews (99,224 rows × 7 columns): Contains customer feedback for each order, including review score, review comments, and review timestamps.
5. Customers (99,441 rows × 5 columns): Contains customer identifiers and location details, including unique customer ID, ZIP code prefix, city, and state.
6. Sellers (3,095 rows × 4 columns): Contains seller identifiers and location details, including ZIP code prefix, city, and state.
7. Products (32,951 rows × 9 columns): Contains product attributes, including category, name and description lengths, number of photos, weight, and dimensions.
8. Geolocation (geo) (1,000,163 rows × 5 columns): Contains ZIP code prefixes and geographical information, including latitude, longitude, city, and state.
9. Category Translation (cat_en) (71 rows × 2 columns): Maps Portuguese product category names to their English equivalents for analysis.